### DISCUSSION TITLE
Add child needs as extra_links from parent, filtered by type

### DISCUSSION THREAD

**Discussion by HughG** [useblocks/sphinx-needs#927] [Q&A]
I'm using Sphinx-Needs 1.2.2.  I'd like to have related needs optionally nested within a parent, but I'd like them to be listed under specific extra_links options, according to their `type`.

I've managed to make a dynamic function to pick out the IDs of child needs for a given type.  However, it seems that dynamic functions can't be used in extra_links options.  If I try the following, I get an error.

```rst
.. fr:: User can confirm that generated data has been queued for export.
    :id: Export_Queue_Status_Visible
    :rationale: If the user cannot see that data is being exported, they may waste time trying to check, or waste time and resources by trying to manually export the data again.
    :design_ideas: [[child_needs_of_type("di")]]

    .. di:: Show the status in the :need:`WorkList`.
        :id: Export_Queue_Status_Visible_InWorkList
```

The error is
```
sphinx.errors.SphinxError: Parsing function string failed for need Export_Queue_Status_Visible: copy('id', '[[child_needs_of_type("di"). EOL while scanning string literal (<unknown>, line 1)
```

Is there some way to make this work, or a better approach to the original goal?

**Comment by danwos** [comment]
Mhh, this should work and it looks like a bug for me, as the error message contains `` copy('id', '[[child_needs_of_type("di")``, 
where the closing ``)]]`` is missing.

Can't say what in details is going wrong here.
Is it possible to share also the code of  your dynamic function ``child_needs_of_type``.
Just to understand how everything fits together.

Thanks for the report  :+1:

**Reply by HughG** [reply]
Hi again, @danwos :-)  Sorry for the very-belated reply.  My function is just this:

```python
def child_needs_of_type(app, need, needs, type, *args, **kwargs):
    for other_need_id, other_need in needs.items():
        if other_need["type"] == type and other_need["parent_need"] == need["id"]:
            return other_need_id
    return "some data"
```
